lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php

```diff
@@ -38,6 +40,23 @@ public function __construct(array $paths)
         parent::__construct(new AttributeReader(), $paths);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function isTransient($className)
+    {
+        $classAnnotations = $this->reader->getClassAnnotations(new ReflectionClass($className));
+
+        foreach ($classAnnotations as $a) {
+            $annot = is_array($a) ? $a[0] : $a;
```

Comment:
@ph-fritsche can `$a` really be an array?

Reply:
```php
            $annot = /* is_array($a) ? $a[0] : */ $a;
```
results in `phpunit` reporting:
```
There was 1 error:

1) Doctrine\Tests\ORM\Mapping\AttributeDriverTest::testIsTransientWithRepeatableAttributes
TypeError: get_class(): Argument #1 ($object) must be of type object, array given

/workspaces/misc/doctrine-orm/lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:52
/workspaces/misc/doctrine-orm/tests/Doctrine/Tests/ORM/Mapping/AttributeDriverTest.php:85
```
due to:
https://github.com/doctrine/orm/blob/332c314e9f055d729555f2861fcb81ee50e482b5/lib/Doctrine/ORM/Mapping/Driver/AttributeReader.php#L26-L28
https://github.com/doctrine/orm/blob/332c314e9f055d729555f2861fcb81ee50e482b5/lib/Doctrine/ORM/Mapping/Driver/AttributeReader.php#L66-L85

Reply:
Also, the phpdoc of `Doctrine\Common\Annotations\Reader` is wrong, and that's in another package :grimacing:

Reply:
Or rather `AttributeReader` does not implement the interface it says it implements. Widening the return type in the interface wouldn't be a BC break for classes implementing it, but it would be a BC break for calling code. Not sure what the best course of action would be here @beberlei . I think we should widen it in `doctrine/annotations`, but to be clean, that would require a major release :grimacing: or since it's just phpdoc, we pray it does not break to many static analysis builds and release it as a minor?

Reply:
Maybe `Doctrine\ORM\AttributeDriver` should not extend `Doctrine\Persistence\Mapping\Driver\AnnotationDriver` at all.
How about we internally refactor `Doctrine\Persistence\Mapping\Driver\AnnotationDriver::getAllClassNames` into another class that could be reused by `Doctrine\ORM\AttributeDriver`?
This way `doctrine/persistence` and `doctrine/annotation` would not have any breaking changes.

If you want to be sentimental about the version increment on this package, one could argue that the declaration of `Doctrine\ORM\AttributeDriver` was broken from the beginning in `2.9` so breaking that inheritance here might be released as a fix.

Reply:
That is a course of action we could take for the next minor release. To fix this bug the small solution plus suppression of tooling errors is my preference.

Reply:
I just thought of another quite simple solution that does not imply changing other packages: let's use an iterable object instead of an array to represent a bug of attributes. As a bonus it will make the code clearer and will answer `can $a really be an array?` because you will have `$a instanceof RepeatableAttributeCollection` instead of `is_array($a)`. WDYT?

Reply:
Here is what it would look like: https://github.com/ph-fritsche/doctrine-orm/pull/1